feat: Introduce Organization Domains (#365)
clerkinc/clerk-docs: docs/references/javascript/organization/domains.mdx

---
title: Organization Domains
description: These are all methods on the Organization class that allow you to manage the domains of an organization.
---

# Organization Domains

These are all methods on [the Organization class](/docs/references/javascript/organization/organization) that allow you to manage the domains of an organization.

## `getDomains()`

Retrieves a list of domains for the currently active organization.

```typescript
function getDomains(params?: GetDomainsParams): Promise<PaginatedResources<OrganizationDomain>>;
```

### `GetDomainsParams`

| Name | Type | Description |
| --- | --- | --- |
| `initialPage?` | `number` | A number that can be used to skip the first n-1 pages. For example, if `initialPage` is set to 10, it is will skip the first 9 pages and will fetch the 10th page. |
| `pageSize?` | `number | A number that indicates the maximum number of results that should be returned for a specific page. |

> | `enrollmentMode?` | `'manual_invitation' \| 'automatic_invitation' \| 'automatic_suggestion'` | The mode in which a domain should operate under |

Reviewer: ```suggestion
| `enrollmentMode?` | `'manual_invitation' \| 'automatic_invitation' \| 'automatic_suggestion'` | An [enrollment mode][enrollment-mode-ref] will change how new users join an organization. |
```

### Returns

> <Tables
>     headings={["Type", "Description"]}
>     headingsMeta={[{}, {maxWidth: "300px"}]}
>     rows={[
>         {cells: [
>                 <code>Promise{"<"}<a href="/docs/references/javascript/types/paginated-response">PaginatedResponse</a>{"<"}<a href="/docs/references/javascript/organization-domain">OrganizationDomain</a>{">"}</code>,
>                 <>This method returns a <code>Promise</code> which resolves with a list of <code><a href="/docs/references/javascript/organization-domain">OrganizationDomain</a></code> objects.</>,
>             ]},
>     ]}
> />

Reviewer: ```suggestion
| Type | Description |
| --- | --- |
| <code>Promise\<[PaginatedResponse][pag-res-ref]\<[OrganizationDomain][org-domain-ref]\>></code> | This method returns a `Promise` which resolves with a list of [`OrganizationDomain`][org-domain-ref] objects. |
```

| Type | Description |
| --- | --- |

> | <code>Promise{"<"}<a href="/docs/references/javascript/types/paginated-response">PaginatedResponse</a>{"<"}<a href="/docs/references/javascript/organization-domain">OrganizationDomain</a>{">"}</code> | This method returns a <code>Promise</code> which resolves with a list of <code><a href="/docs/references/javascript/organization-domain">OrganizationDomain</a></code> objects. |

Reviewer: ```suggestion
| <code>Promise\<[PaginatedResponse][pag-res-ref]\<[OrganizationDomain][org-domain-ref]\>></code> | This method returns a `Promise` which resolves with a list of [`OrganizationDomain`][org-domain-ref] objects. |
```

## `getDomain()`

Retrieves a domain for an organization based on the given domain ID.

```typescript
function getDomain(params: GetDomainParams): Promise<OrganizationDomain>;
```

### `GetDomainParams`

| Name | Type | Description |
| --- | --- | --- |

> | `domainId` | string |  The id of the domain that will be fetched. |

Reviewer: ```suggestion
| `domainId` | string |  The ID of the domain that will be fetched. |
```

### Returns

| Type | Description |
| --- | --- |

> | <code>Promise{"<"}<a href="/docs/references/javascript/organization-domain">OrganizationDomain</a>{">"}</code> | This method returns a <code>Promise</code> which resolves to the <code><a href="/docs/references/javascript/organization-domain">OrganizationDomain</a></code> for the corresponding ID. |

Reviewer: ```suggestion
| <code>Promise\<[OrganizationDomain][org-domain-ref]\></code> | This method returns a `Promise` which resolves to the [`OrganizationDomain`][org-domain-ref] for the corresponding ID. |

[org-domain-ref]: /docs/references/javascript/organization-domain
[pag-res-ref]: /docs/references/javascript/types/paginated-response
[enrollment-mode-ref]: /docs/organizations/verified-domains#enrollment-modes
```

---
title: Organization Domains
description: These are all methods on the Organization class that allow you to manage the domains of an organization.
---

> import { Tables } from "@/components/Table";

Reviewer: We don't need to import components in MDX files, because our marketing repo is rendering our docs repo, and the components exist in the marketing repo.```suggestion
```

# Organization Domains

These are all methods on [the Organization class](/docs/references/javascript/organization/organization) that allow you to manage the domains of an organization.

> ## `getDomains()`

Reviewer: ```suggestion
## `getDomains()`

Retrieves a list of domains for the currently active organization.
```

```typescript
function getDomains(params?: GetDomainsParams): Promise<PaginatedResources<OrganizationDomain>>;
```

### `GetDomainsParams`

| Name | Type | Description |
| --- | --- | --- |

> | `initialPage?` | number | A number that can be used to skip the first n-1 pages. For example, if `initialPage` is set to 10, it is will skip the first 9 pages and will fetch the 10th page. |
> | `pageSize?` | number | A number that indicates the maximum number of results that should be returned for a specific page. |
> | `enrollmentMode?` | `'manual_invitation' \| 'automatic_invitation' \| 'automatic_suggestion'` | The mode in which a domain should operate under |

Reviewer: ```suggestion
| `initialPage?` | `number` | A number that can be used to skip the first n-1 pages. For example, if `initialPage` is set to 10, it is will skip the first 9 pages and will fetch the 10th page. |
| `pageSize?` | `number` | A number that indicates the maximum number of results that should be returned for a specific page. |
| `enrollmentMode?` | `'manual_invitation' \| 'automatic_invitation' \| 'automatic_suggestion'` | The mode in which a domain should operate under. |
```

### Returns

> <Tables

Reviewer: Markdown table

    headings={["Type", "Description"]}
    headingsMeta={[{}, {maxWidth: "300px"}]}
    rows={[
        {cells: [
                <code>Promise{"<"}<a href="/docs/references/javascript/types/paginated-response">PaginatedResponse</a>{"<"}<a href="/docs/references/javascript/organization-domain">OrganizationDomain</a>{">"}</code>,
                <>This method returns a <code>Promise</code> which resolves with a list of <code><a href="/docs/references/javascript/organization-domain">OrganizationDomain</a></code> objects.</>,
            ]},
    ]}
/>

> ## `getDomain()`

Reviewer: ```suggestion
## `getDomain()`

Retrieves a domain for an organization based on the given domain ID.
```

```typescript
function getDomain(params: GetDomainParams): Promise<OrganizationDomain>;
```

> Creates and sends an invitation to the target email address for becoming a member with the role passed on the  function parameters.

Reviewer: The method is getDomain() but takes about creating a domain? Is there a `createDomain()` method missing??

```suggestion
Creates and sends an invitation to the target email address for becoming a member with the role passed on the function parameters.
```

### `GetDomainParams`

| Name | Type | Description |
| --- | --- | --- |

> | `domainId` | string |  The id of the domain that will be fetched. |

Reviewer: ```suggestion
| `domainId` | string |  The ID of the domain that will be fetched. |
```

### Returns

> <Tables

Reviewer: Markdown table
